src/test_fixture/input/mod.rs

```diff
@@ -31,60 +31,66 @@ pub struct GenericReportTestInput<F: Field, MK: BitArray, BK: BitArray> {
 
 #[macro_export]
 macro_rules! ipa_test_input {
-    ( [ $({ match_key: $mk:expr, is_trigger_report: $itr:expr, breakdown_key: $bdk:expr, trigger_value: $tv:expr }),* ]; ($field:tt, $mk_bit_array:tt, $bk_bit_array:tt) ) => {
+    ( { match_key: $mk:expr, is_trigger_report: $itr:expr, breakdown_key: $bk:expr, trigger_value: $tv:expr $(,)? }; ($field:tt, $mk_bit_array:tt, $bk_bit_array:tt) ) => {
+        GenericReportTestInput {
+            match_key: Some(<$mk_bit_array as $crate::bits::BitArray>::truncate_from(u128::try_from($mk).unwrap())),
+            attribution_constraint_id: None,
+            timestamp: None,
+            is_trigger_report: Some($field::from(u128::try_from($itr).unwrap())),
+            breakdown_key: <$bk_bit_array as $crate::bits::BitArray>::truncate_from(u128::try_from($bk).unwrap()),
+            trigger_value: $field::from(u128::try_from($tv).unwrap()),
+            helper_bit: None,
+            aggregation_bit: None,
+        }
+    };
+
+    ( [ $({ match_key: $mk:expr, is_trigger_report: $itr:expr, breakdown_key: $bk:expr, trigger_value: $tv:expr $(,)? }),* $(,)? ]; ($field:tt, $mk_bit_array:tt, $bk_bit_array:tt) ) => {
         vec![
-            $(
-                GenericReportTestInput {
-                    match_key: Some(<$mk_bit_array as $crate::bits::BitArray>::truncate_from(u128::try_from($mk).unwrap())),
-                    attribution_constraint_id: None,
-                    timestamp: None,
-                    is_trigger_report: Some($field::from(u128::try_from($itr).unwrap())),
-                    breakdown_key: <$bk_bit_array as $crate::bits::BitArray>::truncate_from(u128::try_from($bdk).unwrap()),
-                    trigger_value: $field::from(u128::try_from($tv).unwrap()),
-                    helper_bit: None,
-                    aggregation_bit: None
-                }
-            ),*
+            $(ipa_test_input!({ match_key: $mk, is_trigger_report: $itr, breakdown_key: $bk, trigger_value: $tv }; ($field, $mk_bit_array, $bk_bit_array))),*
```

Comment:
Added a new macro rule that generates a single input row. The existing macro rule that generates a vector of rows, now calls this new rule.